### PR TITLE
Use markdown-backed service updates

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

We want to add markdown-backed service updates, added by non-developers.

The request is to provide design-history-esque updates, which are added via markdown files with front matter metadata.

## Changes proposed in this pull request

- Add a `MarkdownDocument` class to load and parse front matter metadata from markdown files
- Add markdown files for content
- Move Placements `render_markdown` helper method into our shared top-level namespace
- Add the `/service-updates` page

## Guidance to review

- Visit `/service-updates`
- Content is still pending.

## Link to Trello card

https://trello.com/c/cO0hIplh/592-updates-page

## Screenshots

CleanShot 2024-07-29 at 15 32 53

### CODE REVIEW

File: lib/markdown_document.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class MarkdownDocument
+  YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
```
Comment (ollietreend) [inline lib/markdown_document.rb:2]: It makes me sad that Redcarpet doesn't handle Markdown frontmatter 😢

File: lib/markdown_document.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  attr_reader :raw_content
+
+  def initialize(raw_content)
+    @raw_content = raw_content
+  end
+
+  def front_matter
+    YAML.safe_load(@raw_content.match(YAML_FRONT_MATTER_REGEXP).to_s, permitted_classes: [Date])
```
Comment (ollietreend) [inline lib/markdown_document.rb:11]: You may want to consider memoizing this value considering it's used multiple times in a request (e.g. for sorting the documents, then multiple times in the view for rendering individual attributes, etc.)

File: lib/markdown_document.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  delegate :[], to: :front_matter
+
+  def content
+    @raw_content.gsub(YAML_FRONT_MATTER_REGEXP, "")
```
Comment (ollietreend) [inline lib/markdown_document.rb:17]: It probably wouldn't hurt to also memoize this value. Although under it's current use case, it seems to only used once per request.

File: app/views/claims/service_updates/content/claim-window-closed.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+title: Claim window closed on 19 July
+date: 2024-07-20
+start_page: true
+---
+
+# Claim window closed on 19 July
+
+20 July 2024
+
+Mud minnow featherfin knifefish brook lamprey Ganges shark gombessa Kafue pike. Butterflyfish orangestriped triggerfish European eel straptail moray eel green swordtail pompano half-gill, sand tilefish roosterfish.
```
Comment (ollietreend) [inline app/views/claims/service_updates/content/claim-window-closed.md:11]: Is this... real content? 🤔 _Should this_ be real content? 😅

**Review by ollietreend — COMMENTED** [review]
Nice re-use of the existing Markdown rendering.

It's a shame the GovukMarkdown / Redcarpet gems don't support frontmatter natively 😢 but this implementation looks sensible. Nice work.

File: lib/markdown_document.rb (hunk 1 above)
Comment (Nitemaeric) [reply lib/markdown_document.rb:2]: Right?!

File: app/views/claims/service_updates/content/claim-window-closed.md (hunk 4 above)
Comment (Nitemaeric) [reply app/views/claims/service_updates/content/claim-window-closed.md:11]: No. This shouldn't be merged yet. Pending real content.

CleanShot 2024-07-30 at 13 16 25

😉

**Comment by Nitemaeric** [comment]
[Updates page](https://trello.com/c/cO0hIplh/592-updates-page)

**Review by ollietreend — APPROVED** [review]
Looks good to me 👍🏻

**Comment by Nitemaeric** [comment]
[Updates page](https://trello.com/c/r0s3ZO4l/621-updates-page)

**Comment by Nitemaeric** [comment]
[Updates page](https://trello.com/c/S5nKeK6i/624-updates-page)

**Pull request merged**